src/client/protocolgamesend.cpp

```diff
@@ -70,19 +70,20 @@ void ProtocolGame::sendLoginPacket(uint challengeTimestamp, uint8 challengeRando
         msg->addU8(0);
 
     const int offset = msg->getMessageSize();
-    // first RSA byte must be 0
-    msg->addU8(0);
 
     if (g_game.getFeature(Otc::GameLoginPacketEncryption)) {
+        // first RSA byte must be 0
+        msg->addU8(0);
```

Comment:
arent you breaking other protocols by this change?

Reply:
with the two bytes i moved around, it works out to be the exact same end result.
The change is simply for a more accurate code flow.
With the current code, the client "expects" the RSA header (which is hardcoded to be read as 0) whether encryption is enabled or not. While the gm byte is inside the encryption feature check (again hardcoded to read 0), while earlier protocols still look for it despite not supporting encryption.

I had also tested this change with 8.00 protocol, and it worked fine.